peer.go

```diff
@@ -306,11 +306,34 @@ func (p *peer) loadActiveChannels(chans []*channeldb.OpenChannel) error {
 		// Register this new channel link with the HTLC Switch. This is
 		// necessary to properly route multi-hop payments, and forward
 		// new payments triggered by RPC clients.
-		sphinxDecoder := htlcswitch.NewSphinxDecoder(p.server.sphinx)
 		link := htlcswitch.NewChannelLink(
 			htlcswitch.ChannelLinkConfig{
-				Peer:             p,
-				DecodeOnion:      sphinxDecoder.Decode,
+				Peer:                  p,
+				DecodeHopIterator:     p.server.sphinx.DecodeHopIterator,
+				DecodeOnionObfuscator: p.server.sphinx.DecodeOnionObfuscator,
+				ChannelEdgeInfo: func(chanID lnwire.ChannelID) (*channeldb.ChannelEdgePolicy,
```

Comment:
As the switch only needs to obtain the latest channel update from _us_, we can simplify this function and the handling within the switch a good bit. Instead of accepting the `chanID` here, we'll just thread a function through that only return the latest channel update. This will let us not have the logic of parsing the feature bits of an update to decide which direction should be returned.